client/src/components/EditProfileButton.js

```diff
@@ -0,0 +1,39 @@
+import React, { useState, useContext } from "react";
+import Dialog from "common/Dialog";
+import EditProfileForm from "components/EditProfileForm";
+import { Button } from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import { Edit } from "@material-ui/icons";
+import { Context as UserContext } from "contexts/AuthContext";
+
+const EditProfileButton = () => {
+  const classes = useStyles();
+  const { updateUser } = useContext(UserContext);
+  const [dialogOpen, setDialogOpen] = useState(false);
+
+  const handleSubmit = async (profileData) => {
+    updateUser(profileData);
+    setDialogOpen(false);
+  };
+
+  return (
+    <>
+      <Button
+        variant="outlined"
+        size="large"
+        onClick={() => setDialogOpen(true)}
+      >
+        <Edit />
+        Edit
+      </Button>
+
+      <Dialog isOpen={dialogOpen} handleClose={() => setDialogOpen(false)}>
+        <EditProfileForm onSubmit={handleSubmit} />
+      </Dialog>
```

Comment:
I think as Kareem mentioned in sprint planning might be better to wrap the dialogs in a context as opposed to putting them right in the Button component - or I'd expect them to exist at a more parent component (maybe wrapped a in route so that you can have it appear when you have a certain route open)

By doing this current approach, we are creating a lot of extra React components that are unnecessary and harder to find as well - especially since they are in components you would not expect